libp2p/protocols/rendezvous.nim

```diff
@@ -506,24 +508,35 @@ proc advertisePeer(rdv: RendezVous, peer: PeerId, msg: seq[byte]) {.async.} =
   await rdv.sema.acquire()
   discard await advertiseWrap().withTimeout(5.seconds)
 
-method advertise*(
-    rdv: RendezVous, ns: string, ttl: Duration = MinimumDuration
-) {.async, base.} =
-  let sprBuff = rdv.switch.peerInfo.signedPeerRecord.encode().valueOr:
-    raise newException(RendezVousError, "Wrong Signed Peer Record")
+proc batchAdvertise*(
```

Comment:
I have just realized this has been changed from a method to a proc, could you please explain why?

Reply:
The `advertise` method now call a public `batchAdvertise` proc so that the interface still work like before but you can also use the proc with the new parameters.

Reply:
@lchenut does it make sense to pass the peers as a parameter here instead of using the seq inside the rdv instance?

Reply:
There's really nothing against it. In theory, the rendezvous protocol do it with the peer event handler on joined/left, but if you want to use your own, why not. It wasn't something we needed at the beginning (because we don't really use it outside the test), but if waku needs it, it's a good thing to implement.

Reply:
Ideally, it would be good to add docs explaining the use of those procs and testing covering them. I don't know if different names for them are a good idea either.

Reply:
> Ideally, it would be good to add docs explaining the use of those procs and testing covering them. I don't know if different names for them are a good idea either.

How would I update the docs? Is there a different repo?

Tests already call the new procs via the methods so it is covered as for names I'm open to suggestions.

Reply:
@lchenut could you please create an issue to add docs to the procs?

Reply:
@SionoiS the docs are added right below the proc definition. For the name, I'd just use `advertise` as both procs do the same thing.

Reply:
@lchenut it'd be good to add to the issue an evaluation of the use of methods and procs.

Reply:
I'm not sure I understand your last point

Reply:
Some things are implemented as procs and some as methods, why?

Reply:
I did a little checking. The Rendezvous protocol inherits from LPProtocol, so it should have a `start` and `stop` method, which it does. But `advertise` is also a method, which is wrong, it should be proc.

Reply:
This explains the difference and when to use one or the other https://matthiashager.com/proc-method-nim